app/os_macos.go

```diff
@@ -557,6 +562,22 @@ func gio_onMouse(view, evt C.CFTypeRef, cdir C.int, cbtn C.NSInteger, x, y, dx,
 	})
 }
 
+//export gio_onExternalDrop
+func gio_onExternalDrop(view C.CFTypeRef, path *C.char) {
+	fileUrl := C.GoString(path)
+	w := mustView(view)
+
+	fileExtension := filepath.Ext(fileUrl)
+	mime := mime.TypeByExtension(fileExtension)
+
+	w.w.Event(transfer.DataEvent{
+		Type: mime,
+		Open: func() io.ReadCloser {
+			return io.NopCloser(strings.NewReader(fileUrl))
+		},
+	})
```

Comment:
Hm, thinking of this some more, it seems better to present the data directly, instead of through an URL. Otherwise, how would a receiver tell the difference between content of type `Type` and a file URL pointing to a file of type `Type`? If you need the filename it seems ok to add that as a field on `transfer.DataEvent`.

However, that also means you need to add an `error` to the `transfer.DataEvent.Open` field.

Reply:
Another correct solution would be setting `Type` to `"text/uri-list"`, and then returning `file://` data through the `Open` reader.

Reply:
Less magic, sure, but also more work for every dnd target that has to implement support for "text/uri-list". Your browser example is compelling, though.